File: examples/write_your_own_validator/example_both/validator.js
/** Example both-languages validator (JS) — for local parity CI only. */

export class ExampleBothValidator {
  name = "example-both";
  latencyClass = "inline";

  async index(_message, _context) {
    return;
  }

  judge(message, _context) {
    const score = Math.min(1.0, message.content.length / 100.0);
    const label = score >= 0.3 ? "ok" : "low";
    return { score, label, rationale: "length-based" };
  }
}

File: examples/write_your_own_validator/example_both/validator.py
"""Example both-languages validator (Python) — for local parity CI only.

This is a TEMPLATE/fixture, not a published third-party validator.
"""

from __future__ import annotations


class ExampleBothValidator:
    name = "example-both"
    latency_class = "inline"

    async def index(self, message, context) -> None:
        return None

    def judge(self, message, context):
        score = min(1.0, len(message.content) / 100.0)
        # Match the TS implementation exactly
        label = "ok" if score >= 0.3 else "low"
        return {
            "score": score,
            "label": label,
            "rationale": "length-based",
        }
